src/jit/importer.cpp

```diff
@@ -17565,6 +17565,14 @@ void Compiler::impMarkInlineCandidate(GenTreePtr             callNode,
     GenTreeCall* call = callNode->AsCall();
     InlineResult inlineResult(this, call, nullptr, "impMarkInlineCandidate");
 
+#ifdef _TARGET_ARM_
+    // When inlining methods that return type is struct, 
+    // legacy JIT for ARM fails to assign correct value for struct return
+    if (call->gtReturnType == TYP_STRUCT) {
```

Comment:
@wateret I didn't checked this inlining issue occur when JIT'ed code passed struct value using physical register(s). But this problem occur for passing struct not using registers to pass struct value.

If this problem occur on HFA struct as well, we also need to avoid inlining all methods that return strcut type. And this commit can do that.